src/services/DataExportService.ts

```diff
@@ -0,0 +1,163 @@
+/**
+ * DataExportService
+ * Handles inventory data export and sharing functionality
+ */
+
+import type { InventoryItems } from '../inventory.js';
+
+export interface ExportData {
+  version: string;
+  timestamp: string;
+  exportedAt: string;
+  inventory: InventoryItems;
+  metadata: {
+    totalColors: number;
+    colorsWithStock: number;
+    totalQuantity: number;
+  };
+}
+
+export interface ExportOptions {
+  filename?: string;
+}
```

Comment:
At this point, it doesn't make sense to pass the filename from the consumer.
We can simplify further, and always compute the filename without any possibility of customization.